app/controllers/campaigns_controller.rb

```diff
@@ -0,0 +1,40 @@
+class CampaignsController < ApplicationController
+	def index
+		@campaigns = Campaign.all
+	end
+
+	def create
+		@campaign = Campaign.new(campaign_params)
+		if @campaign.save
+			redirect_to @campaign
+		else
+			render 'new'
+		end
+	end
+
+	def new
+		@campaign = Campaign.new
+	end
+
+	def show
+		@campaign = Campaign.find(params[:id])
+	end
+
+	def edit
+		@campaign = Campaign.find(params[:id])
+	end
+
+	def update
+		@campaign = Campaign.find(params[:id])
+		if @campaign.update(campaign_params)
+			redirect_to @campaign
+		else
+			render "edit"
+		end
+	end
+
+	private
```

Comment:
Oh. Right. Gotcha. That makes sense. Language/framework context switching is hard :)